homeassistant/components/soma/cover.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for Soma Covers.
+
+For more details about this platform, please refer to the documentation at
+"""
+
+import logging
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION
+from homeassistant.components.soma import DOMAIN, SomaEntity, DEVICES, API
+from homeassistant.exceptions import PlatformNotReady
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Soma cover platform."""
+    def get_covers():
+        """Retrieve covers."""
+        devices = hass.data[DOMAIN][DEVICES]
+
+        return [SomaCover(cover, hass.data[DOMAIN][API]) for cover in
+                devices]
+
+    async_add_entities(await hass.async_add_executor_job(get_covers), True)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Old way of setting up platform.
+
+    Can only be called when a user accidentally mentions the platform in their
+    config. But even in that case it would have been ignored.
+    """
+    pass
+
+
+class SomaCover(SomaEntity, CoverDevice):
+    """Representation of a Soma cover device."""
+
+    async def async_update(self):
+        """Update the device with the latest data."""
+        await super().async_update()
```

Comment:
If we don't do anything special in the child method we don't need to overwrite the parent method.

Reply:
Fixed

Reply:
This still needs to be addressed.